Source/Basetypes/BigFloat.cs

```diff
@@ -157,6 +157,57 @@ public static BigFloat FromBigDec(BigDec v, int significandSize, int exponentSiz
       return new BigFloat(isNeg, sig, exp, sigSize, expSize);
     }
 
+    [Pure]
+    public static BigFloat FromBVString(String s)
+    {
+      /*
+       * String must be either of the format (fp (_ bv* *) (_ bv* *) (_ bv* *))
+       * or of the special value format: (_ val * *)
+       * Where * indicates an integer value
+       * and val is one of the special values: NaN, +oo, -oo, zero, -zero
+       */
+
+      if (s.IndexOf('_') == 1)
+      {
+        s = s.Substring(0, s.Length - 1);
+        String[] args = s.Split(' ');
+        int sigSize = int.Parse(args[3]);
```

Comment:
Do you mean replace that line with something like this:

int sigSize = Convert.ToInt32(uint.Parse(args[3]));

The actual variable can't be of type 'uint' because the BigFloat constructor expects integers for the sigSize and expSize parameters. Should this be changed?

Reply:
Interesting. I think the current implementation is fine. There's no need to change this.